src/marks/auto.d.ts

```diff
@@ -3,26 +3,74 @@ import type {CompoundMark, Data} from "../mark.js";
 import type {Reducer} from "../reducer.js";
 import type {BinOptions} from "../transforms/bin.js";
 
+// TODO: should jsdoc go here or inside interface?
+type MarkType = "dot" | "line" | "area" | "rule" | "bar";
+
 export interface AutoOptions {
+  /**
+   * The horizontal spatial encoding. If *y* is set without *x*, the *x* channel
+   * will default to the **count** reducer to produce a histogram; you can
+   * override this by passing {reduce: null}.
+   */
   x?: ChannelValue | Reducer | ({value?: ChannelValue; reduce?: Reducer | null; zero?: boolean} & BinOptions);
+  /**
+   * The vertical spatial encoding. If *x* is set without *y*, the *y* channel
+   * will default to the **count** reducer to produce a histogram; you can
+   * override this by passing {reduce: null}.
+   */
   y?: ChannelValue | Reducer | ({value?: ChannelValue; reduce?: Reducer | null; zero?: boolean} & BinOptions);
+  /**
+   * The color encoding, which corresponds to the stroke channel for dots,
+   * lines, and rules, and the fill channel for areas and bars.
+   */
   color?: ChannelValue | Reducer | {value?: ChannelValue; reduce?: Reducer | null; color?: string};
+  /**
+   * The size encoding corresponds to the radius channel of dots; it may
+   * correspond to the length of vectors in the future.
+   */
   size?: ChannelValue | Reducer | {value?: ChannelValue; reduce?: Reducer | null};
+  /**
+   * The horizontal facet dimension.
+   */
   fx?: ChannelValue | {value?: ChannelValue};
+  /**
+   * The vertical facet dimension.
+   */
   fy?: ChannelValue | {value?: ChannelValue};
-  mark?: "dot" | "line" | "area" | "rule" | "bar";
+  /**
+   * A type of mark; for example, the type **bar** encompasses bar, barX, and
+   * barY. Used to guide what sort of mark *auto* should use. It should be thought
```

Comment:
```suggestion
   * A type of mark; for example, the type *bar* encompasses bar, barX, and
   * barY. Used to guide what sort of mark **auto** should use. It should be thought
```

option (strings) in italics, marks in bold